### PR TITLE
Transition from parking lot to unavailable fors

### PR DESCRIPTION
**Pull request by eanders**

### CODE REVIEW

File: app/controllers/reports/parked_clients_controller.rb [hunk 1]
```diff
@@ -7,20 +7,43 @@
...
-        clients = parked_scope.non_confidential.full_release
-      end
+      @active_tab = params[:tab] || route_to_html_id(@routes.first)
+      clients = parked_scope
+      clients = clients.non_confidential.full_release unless can_view_all_clients?
+
       @clients = clients.
-        order(prevent_matching_until: :asc).
+        joins(:unavailable_as_candidate_fors).
+        order('unavailable_as_candidate_fors.expires_at asc').
```
Comment (awisema) [inline app/controllers/reports/parked_clients_controller.rb]: Add `uacf_t` to `ArelHelper` and replace the string with `uacf_t[:expires_at].asc`?

File: app/models/client.rb [hunk 2]
```diff
@@ -501,11 +500,7 @@ def available_text
         'Fully matched'
       end
     else
-      if parked?
-        "Parked until #{prevent_matching_until}"
-      else
-        'Not available'
-      end
+      'Not available'
```
Comment (awisema) [inline app/models/client.rb]: Maybe bring back the if and return`Not available on any route`? It seems like there is a difference between these two cases that might matter to users.

File: db/migrate/20200312144826_update_existing_unavailable_fors.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+class UpdateExistingUnavailableFors < ActiveRecord::Migration[6.0]
+  def up
+    expiration_length = Config.get(:unavailable_for_length)
+    return if expiration_length.zero?
+
+    UnavailableAsCandidateFor.find_each do |u|
+      u.update(expires_at: u.created_at + expiration_length.days)
+    end
+
+    Client.where("prevent_matching_until > ?", Date.current).find_each do |client|
```
Comment (awisema) [inline db/migrate/20200312144826_update_existing_unavailable_fors.rb]: Seems to me that we should either move all parked client, not just the future ones... If there are expired parked clients, should we be cleaning them up?

Comment (eanders) [reply db/migrate/20200312144826_update_existing_unavailable_fors.rb]: Those should have been cleaned up already.  There's a nightly process that takes care of that currently.  Also, we shouldn't be looking at the field anymore, and it should probably be removed eventually.

File: app/models/client.rb (hunk 2 above)
Comment (eanders) [reply app/models/client.rb]: There is a difference, but we can no longer include a date because each "parking" can include its own date.

Comment (awisema) [reply app/models/client.rb]: Yes, I just wondered if we should return different language so the user might think to look at the routes.. I'm tempted to suggest something like "Not available on any route until {earliest_date}" but that might be misleading...

File: db/migrate/20200312144826_update_existing_unavailable_fors.rb (hunk 3 above)
Comment (awisema) [reply db/migrate/20200312144826_update_existing_unavailable_fors.rb]: If they are cleaned up, can we just do `Client.where.not(prevent_matching_until: nil)`?

**Pull request merged**